library/ssl_misc.h

```diff
@@ -2419,4 +2419,67 @@ int mbedtls_ssl_check_dtls_clihlo_cookie(
                            unsigned char *obuf, size_t buf_len, size_t *olen );
 #endif
 
+#if defined(MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED)
+/* Check if we have any PSK to offer, returns 0 if PSK is available.
+ * Assign the psk and ticket if pointers are present.
+ */
+static inline int mbedtls_ssl_get_psk_to_offer(
```

Comment:
What's the intention of this re-location?

Reply:
It should be moved into generic. 

For this PR, it should be Okay to get psk from `ssl->conf` . But `psk` configuration code has been changed in `development` .   It will call PSA functions to get psk, it is too heavy to define the function  as `static inline`